src/aiy/cloudspeech.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An API to access Google Speech recognition service."""
+
+import os.path
+
+import aiy._apis._speech
+import aiy.audio
+import aiy.voicehat
+
+# Global variables. They are lazily initialized.
+cloudspeech_recognizer = None
+
+# Expected location of the CloudSpeech credentials file:
+CLOUDSPEECH_CREDENTIALS_FILE = os.path.expanduser('~/cloud_speech.json')
+
+class _CloudSpeechRecognizer(object):
+  """A speech recognizer backed by the Google CloudSpeech APIs.
+  """
+
+  def __init__(self, credentials_file):
+    self._request = aiy._apis._speech.CloudSpeechRequest(credentials_file)
+    self._recorder = aiy.audio.get_recorder()
+
+  def recognize(self):
+    """Recognizes the user's speech and transcript it into text.
+
+    This function listens to the user's speech via the VoiceHat speaker. Then it
+    contacts Google CloudSpeech APIs and returns a textual transcript if possible.
+    """
+    self._request.reset()
+    self._request.set_endpointer_cb(self._endpointer_callback)
+    self._recorder.add_processor(self._request)
+    return self._request.do_request().transcript
+
+  def expect_phrase(self, phrase):
```

Comment:
wouldn't it be nicer to accept a list of phases via *args. Then one can write:
recognizer.expect_phrases('light on', 'light off')